pywbem_mock/_mainprovider.py

```diff
@@ -0,0 +1,4209 @@
+
+    def get_class_store(self, namespace):
+        """
+        Returns the class object store for the specified CIM namespace within
+        the CIM repository.  This method validates that the namespace exists in
+        the data store.
+
+        This method does not validate that the namespace exists. Be certain
+        namespace is validated against CIM repository before calling this
+        method
+
+        Parameters:
+
+          namespace(:term:`string`):
+            The name of the CIM namespace in the CIM repository (case
+            insensitive). Must not be `None`. Leading or trailing
+            slash charactes are ignored.
+
+        Returns:
+
+          Instance of class derived from :class:`~pywbem_mock.`BaseObjectStore`
+          which is the object store for classes in the CIM repository.
+
+        """
+        return self.cimrepository.get_class_store(namespace)
+
+    def get_instance_store(self, namespace):
+        """
+        Returns the instance object store for the specified CIM namespace
+        within the CIM repository.  This method validates that the namespace
+        exists.
+
+        This method does not validate that the namespace exists. Be certain
+        namespace is validated against CIM repository before calling this
+        method
+
+        Parameters:
+
+          namespace(:term:`string`):
+            The name of the CIM namespace in the CIM repository (case
+            insensitive). Must not be `None`. Leading or trailing
+            slash characters are ignored.
+
+        Returns:
+
+           Instance of CIM Repository class
+           derived:class:`~pywbem_mock.`BaseObjectStore` which is the object
+           store for instances in the CIM repository.
+
+        Raises:
+
+          :exc:`~pywbem.CIMError`: CIM_ERR_INVALID_NAMESPACE: Namespace does
+            not exist.
+        """
+
+        return self.cimrepository.get_instance_store(namespace)
+
+    def get_qualifier_store(self, namespace):
+        """
+        Returns the qualifier declaration object store for the specified CIM
+        namespace within the CIM repository.  This method validates that the
+        namespace exists.
+
+        This method does not validate that the namespace exists. Be certain
+        namespace is validated against CIM repository before calling this
+        method
+
+        Parameters:
+
+          namespace(:term:`string`):
+            The name of the CIM namespace in the CIM repository (case
+            insensitive). Must not be `None`. Leading or trailing
+            slash characters are ignored.
+
+        Returns:
+
+          Instance of CIM Repository class derived
+          from:class:`~pywbem_mock.`BaseObjectStore` which is the object store
+          for qualifier declarations in the CIM repository.
+
+        Raises:
+
+          :exc:`~pywbem.CIMError`: CIM_ERR_INVALID_NAMESPACE: Namespace does
+            not exist.
```

Comment:
If the namespace does not exist, this method now raises `KeyError` in `self.cimrepository.get_qualifier_store()`.

Reply:
Changed to use your sentence, statementt that namespace should be validated before calling this method. Finally added KeyError as raises.

DONE